### PR TITLE
[tools/Depends] Bump openssl 3.0.13

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Bump openssl to LTS 3.0 series

## Motivation and context
Openssl 3 take 2. Cargo-c update is due to our docker builders failing to build with openssl 3

getauxval revert patch is still required. Testing with aarch64 android 14 phone had the following errors from regular traffic (eg install addon from repos) without the patch

```
CCurlFile::CReadState::FillBuffer - (0xb4000073facc43100) Failed: SSL connect error (35)
```

I do have a branch that brings it to 3.1.5, and we should be able to go to 3.2/3.3 pretty easily as far as i can see from openssl 1 to 3 migration guidance docs, but 3.0 is LTS and has the longest support runway out of current point releases.

## How has this been tested?
macos aarch64 runtime
android aarch64 runtime

Addon video playback, repo install addon

Will need/want testing on webos

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/openssl/Makefile [hunk 1]
```diff
@@ -4,15 +4,27 @@ DEPS = ../../Makefile.include Makefile OPENSSL-VERSION ../../download-files.incl
 
 # configuration settings
 ifeq ($(OS), linux)
-  # Need to export our vars to override "default" conf data of openssl
   TARGETOPT=--with-zlib-include=$(PREFIX)/include --with-zlib-lib=$(PREFIX)/lib
-  CONFIGURE=MACHINE=$(PLATFORM) ./config no-shared zlib no-tests no-asm $(TARGETOPT) --prefix=$(PREFIX) --libdir=lib
+  CONFIGURE=./Configure no-shared zlib no-tests no-asm no-module  $(TARGETOPT) --prefix=$(PREFIX) --libdir=lib
+  ifeq ($(TARGET_PLATFORM),webos)
+    CONFIGURE+= linux-armv4
```
Comment (sundermann) [inline tools/depends/target/openssl/Makefile:10]: Do we need the special handling for webOS here? Currently `CPU=armeabi-v7a` on webOS. I looked for usages of `armeabi-v7a` in `tools/depends` and it seems that it's not really used anywhere. I think we can safely set https://github.com/xbmc/xbmc/blob/29e78e56830187b8950afc401404b4f86add7ac4/tools/depends/configure.ac#L363 to `arm` and remove this specific check here.

**Review by sundermann — APPROVED** [review]
Runs fine on webOS.

File: tools/depends/target/openssl/Makefile (hunk 1 above)
Comment (fuzzard) [reply tools/depends/target/openssl/Makefile:10]: I might leave that for you in a follow up. I cant personally test/build, and just dont want to get stuck in a rabbit hole i cant fix.
Your reasoning sounds sane, so fingers crossed its that easy.

File: tools/depends/native/cargo-c/CARGO-C-VERSION [hunk 2]
```diff
@@ -1,5 +1,8 @@
 APPNAME=cargo-c
-VERSION=0.9.21
+VERSION=0.9.31
 SOURCE=$(APPNAME)-$(VERSION)
-ARCHIVE=$(SOURCE).tar.gz
-SHA512=855391c29843f8e5f204f889cab16d5d569ebb2174367a8be0d4be3d87141133f98fb7a6750ed0030783a1ff29f358d60c6ca79ed5bb65f61196c726f9c1a0ec
+#ARCHIVE=$(SOURCE).tar.gz
```
Comment (wsnipex) [inline tools/depends/native/cargo-c/CARGO-C-VERSION:4]: Todo

File: tools/depends/native/cargo-c/CARGO-C-VERSION [hunk 3]
```diff
@@ -1,5 +1,8 @@
...
+ARCHIVE=v$(VERSION).tar.gz
+SHA512=12aa5fd857ca6fdba50ffb08d120084ac063b5dfad037b54caf3e28ccaa36bfd07b43e61619ccce72c081655e1759be05f455ea93caebb0e763a63dd4246d0cf
+
+BASE_URL=https://github.com/lu-zero/cargo-c/archive/refs/tags
```
Comment (wsnipex) [inline tools/depends/native/cargo-c/CARGO-C-VERSION:8]: Todo

**Review by wsnipex — COMMENTED** [review]
Looks good

**Pull request merged**